[CI] SoftDeletesRetentionIT testRetentionLeasesIgnoredForSoftDeletesPolicy failing

**Build Scans:**
- [elasticsearch-periodic #12239 / openjdk21_checkpart2_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/tmwflajc2bako)
- [elasticsearch-periodic #12239 / openjdk21_checkpart2_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/77isoaiu4oryg)
- [elasticsearch-pull-request #143646 / part-2](https://gradle-enterprise.elastic.co/s/ucswkcwo3d2oc)
- [elasticsearch-pull-request #143646 / part-2](https://gradle-enterprise.elastic.co/s/fd6mo46uc2gqy)
- [elasticsearch-pull-request #143243 / part-2](https://gradle-enterprise.elastic.co/s/kt6dubgq773uu)
- [elasticsearch-pull-request #143243 / part-2](https://gradle-enterprise.elastic.co/s/bg3vdc7tt6veg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.engine.SoftDeletesRetentionIT.testRetentionLeasesIgnoredForSoftDeletesPolicy" -Dtests.seed=8EBF01BBC2CFD174 -Dtests.locale=sr-Latn-ME -Dtests.timezone=America/Whitehorse -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.engine.SoftDeletesRetentionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRetentionLeasesIgnoredForSoftDeletesPolicy'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: [rgnlfjvzjpxa][0] _seq_no doc values should be empty
Expected: <2147483647>
     but: was <0>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step openjdk21_checkpart2_java-fips-140-3-matrix
- [main] 6 failures in test testRetentionLeasesIgnoredForSoftDeletesPolicy (1.2% fail rate in 485 executions)
- [main] 2 failures in step openjdk21_checkpart2_java-fips-140-3-matrix (66.7% fail rate in 3 executions)
- [main] 4 failures in step part-2 (1.2% fail rate in 335 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.6% fail rate in 333 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -476,9 +476,6 @@ tests:
 - class: org.elasticsearch.compute.data.BlockAccountingTests
   method: testBytesRefVector
   issue: https://github.com/elastic/elasticsearch/issues/148481
-- class: org.elasticsearch.xpack.stateless.engine.SoftDeletesRetentionIT
-  method: testRetentionLeasesIgnoredForSoftDeletesPolicy
-  issue: https://github.com/elastic/elasticsearch/issues/148557
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/148339

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/engine/SoftDeletesRetentionIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/engine/SoftDeletesRetentionIT.java
@@ -108,8 +108,9 @@ public class SoftDeletesRetentionIT extends AbstractStatelessPluginIntegTestCase
             assertTrue(leases.stream().anyMatch(l -> l.id().equals(retentionLeaseId) && l.retainingSequenceNumber() == 0L));
         });
 
-        // Update random docs to create soft-deleted versions of the originals
-        var randomDocs = randomNonEmptySubsetOf(docsIds);
+        // Update random docs to create soft-deleted versions of the originals.
+        // We leave at least one doc behind, so the merge still has a target to hit.
+        var randomDocs = randomSubsetOf(randomIntBetween(1, docsIds.size() - 1), docsIds);
         var bulk = client().prepareBulk().setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
         for (var randomDoc : randomDocs) {
             bulk.add(prepareIndex(indexName).setId(randomDoc).setSource("field", "updated-" + randomDoc));

